core/templates/dev/head/components/rule_editor_directive.html

```diff
@@ -18,6 +18,9 @@
         <span ng-if="item.type == 'checkboxes'" style="color: black;">
           <object-editor obj-type="SetOfHtmlString" init-args="{choices: ruleDescriptionChoices}" value="rule.inputs[item.varName]"></object-editor>
         </span>
+        <span ng-if="item.type == 'dropdown'" style="color: black;">
+          <object-editor obj-type="ListOfSetsOfHtmlStrings" init-args="{choices: ruleDescriptionChoices}" value="rule.inputs[item.varName]"></object-editor>
```

Comment:
Hi @vibhor98, I think here you need to replace `init-args="{choices: ruleDescriptionChoices}"` with `get-init-args="getInitArgs()"` and then define:
```js
scope.getInitArgs = function() {
   return ..
}
```
in the parent directive.
This seems to be causing a problem here as `ListOfSetsOfHtmlStringsEditor` uses `getInitArgs` [here](https://github.com/oppia/oppia/blob/develop/extensions/objects/templates/ListOfSetsOfHtmlStringsEditor.js#L36) which is not defined. Therefore the error faced by @kevinlee12.

Reply:
Oh, I think I am wrong here -- these are all related to `ObjectEditorDirective`. Therefore your approach should work. I am not sure why it isn't. Sorry!

Reply:
Hi @kevinlee12, As you might have seen I have added special case for `ListOfSetsOfHtmlStrings` object in the `RuleEditorDirective`. `ListOfSetsOfHtmlStringsEditor` receives `getInitArgs` method from its parent `ObjectEditorDirective` which receives the `choices` array from this file. By keeping in mind this hierarchy and following `SetOfHtmlString`, my approach should work. But still it throws `undefined getInitArgs`.